packages/stats-py/aqsha_stats/analyses/uji_wilcoxon.py

```
@@ -0,0 +1,76 @@
+            [f"Negative Ranks ({post} < {pre})", neg],
+            [f"Positive Ranks ({post} > {pre})", pos],
+            ["Ties", ties],
+            ["Total", n],
+        ],
+    )
+    try:
+        w_stat, p = stats.wilcoxon(a, b)
+    except ValueError as exc:
+        from ..contract import AnalysisError
+
+        raise AnalysisError("insufficient_data", f"Uji Wilcoxon tidak bisa dihitung: {exc}")
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)